model/task/task.go

```diff
@@ -158,8 +158,23 @@ type Task struct {
 	ExecutionPlatform ExecutionPlatform `bson:"execution_platform,omitempty" json:"execution_platform,omitempty"`
 
 	// The version of the agent this task was run on.
-	AgentVersion string                 `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
-	TaskOutput   *taskoutput.TaskOutput `bson:"task_output,omitempty" json:"task_output,omitempty"`
+	AgentVersion string `bson:"agent_version,omitempty" json:"agent_version,omitempty"`
+	// TaskOutputInfo holds the information for the interface that
+	// coordinates persistent storage of a task's output data.
+	// There are four possible scenarios:
+	//     1. The task will never have output data (e.g., display tasks)
+	//        and, therefore, the value is and always will be nil.
+	//     2. The task does not have output data yet, but can in the future
+	//        after running, and, therefore, the value is currently nil.
+	//     3. The task ran with the task output information initialized
+	//        upon getting dispatched and the application can safely use
+	//        this field to fetch any output data.
+	//     4. The task has data but was run before the introduction of the
+	//        field and should be initialized to the zero value on the
+	//        application to safely fetch any output data.
+	// This field should *never* be accessed directly, instead call
+	// `Task.getTaskOutputSafe()` to instantiate the task output interface.
```

Comment:
`getTaskOutputSafe` can't used to instantiate this field because it checks for an undispatched task status. Is this supposed to be `initializeTaskOutputInfo` to write the field and `getTaskOutputSafe` to read it?

Reply:
It does instantiate it when appropriate (in cases where the task is undispatched it will return `nil, false` since it cannot instantiate it, hence why it is "safe"). The `initializeTaskOutputInfo` function is really only used for setting up the values to store in the DB upon dispatch.

Reply:
The phrasing is kind of because it talks about accessing (i.e. reading) + instantiating (i.e. writing/giving it an initial value), but they're supposedly different things, right? Even if you call `getTaskOutputSafe`, this field will still remain uninitialized. Maybe say something like "use `getTaskOutputSafe` to access this value, use `initializeTaskOutputInfo` to initialize it"?

Reply:
I ended up just simplifying it:
```
// This field should *never* be accessed directly, instead call
// `Task.getTaskOutputSafe()`.
```
The function definition itself explains how the instantiation works.